simplq/src/components/pages/Admin/QueueHistory.jsx

```diff
@@ -1,13 +1,177 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import HistoryIcon from '@material-ui/icons/History';
 import SidePanelItem from 'components/common/SidePanel/SidePanelItem';
+import { useSelector } from 'react-redux';
+import { selectRemoveTokens, selectTokens } from 'store/selectedQueue';
+import moment from 'moment';
+import AddedInQueue from '../QueueHistory/AddedInQueue';
+import RemovedFromQueue from '../QueueHistory/RemovedFromQueue';
+import styles from './QueueHistory.module.scss';
 
 export default () => {
+  const [tokens, setTokens] = useState([]);
+  const [removedTokens, setRemovedTokens] = useState([]);
+  const [queueHistory, setQueueHistory] = useState([]);
+  const [showMore, setShowMore] = useState(false);
+  const [count, setCount] = useState(0);
+  const [sortedData, setSortedData] = useState([]);
+
+  const added = useSelector(selectTokens);
+  const removed = useSelector(selectRemoveTokens);
+
+  useEffect(() => {
+    (async () => {
+      let lastAdded = [];
+      let lastRemoved = [];
+      let lastRemovedPrime = [];
+      if (added) {
+        lastAdded = await added.filter((item) => {
+          let isPresent = false;
+          tokens.forEach((token) => {
+            if (item.tokenId === token.tokenId) {
+              isPresent = true;
+            }
+          });
+
+          return !isPresent;
+        });
+      }
+      if (removed) {
+        lastRemoved = await removed.filter((item) => {
```

Comment:
why did we need an await here?

Reply:
I was using a different way to solve this thing and later on I changed it into the present code but forgot to remove await. I have removed it now!